### PR TITLE
[Enhancement](routine_load) Support show routine load statement with like predicate

### LINKED ISSUE #8187: [Enhancement] Support show routine load statement with like predicate
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

There are many routine load jobs, support show routine load statement with like predicate will bring more convenience 

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #8187 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (Yes)
5. Are there any changes that cannot be rolled back: (Yes)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/sql-reference/sql-statements/Data Manipulation/SHOW ROUTINE LOAD.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ under the License.
 ## description
     该语句用于展示 Routine Load 作业运行状态
     语法：
-        SHOW [ALL] ROUTINE LOAD [FOR jobName];
+        SHOW [ALL] ROUTINE LOAD [FOR jobName][LIKE pattern];
```
Comment (EmmyMiao87) [inline docs/zh-CN/sql-reference/sql-statements/Data Manipulation/SHOW ROUTINE LOAD.md:31]: Please update english doc at the same place ~

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -1873,21 +1873,21 @@ stop_routine_load_stmt ::=
     ;
 
 show_routine_load_stmt ::=
-    KW_SHOW KW_ROUTINE KW_LOAD KW_FOR job_label:jobLabel
+    KW_SHOW KW_ROUTINE KW_LOAD KW_FOR job_label:jobLabel opt_wild_where
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/cup/sql_parser.cup:1876]: This means 'show routine load for xxx like xxx'.
But I understand what you mean should be for and like pick only one?

Comment (caiconghui) [reply fe/fe-core/src/main/cup/sql_parser.cup:1876]: my fault, I will fix it.

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java [hunk 3]
```diff
@@ -483,7 +484,7 @@ public RoutineLoadJob getJob(String dbFullName, String jobName) throws MetaNotFo
       if includeHistory is false, filter not running job in result
       else return all of result
      */
-    public List<RoutineLoadJob> getJob(String dbFullName, String jobName, boolean includeHistory)
+    public List<RoutineLoadJob> getJob(String dbFullName, String jobName, boolean includeHistory, PatternMatcher matcher)
             throws MetaNotFoundException {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:488]: Please add a check " the 'jobName' and 'pattern' cannot exist at the same time" ~

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:488]: done

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java [hunk 4]
```diff
@@ -483,8 +484,10 @@ public RoutineLoadJob getJob(String dbFullName, String jobName) throws MetaNotFo
...
+        Preconditions.checkArgument(jobName == null || matcher == null,
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:489]: can be both null, but not both not null ~

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:489]: this check mean that jobName and matcher should not both not null at the same time

Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:489]: you are right ~

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**